stdlib/public/stubs/UnicodeNormalization.cpp

```diff
@@ -129,8 +129,8 @@ int32_t _swift_stdlib_unicode_compare_utf16_utf16(const uint16_t *LeftString,
                                                   const uint16_t *RightString,
                                                   int32_t RightLength) {
   return ucol_strcoll(GetRootCollator(),
-    LeftString, LeftLength,
-    RightString, RightLength);
+    (const UChar *)LeftString, LeftLength,
+    (const UChar *)RightString, RightLength);
```

Comment:
I see.  This is complicated.  Could you check that casting `uint16_t *` to `wchar_t *` is valid from the strict aliasing perspective?  We don't want miscompiles.  Otherwise, I'm OK with the change.